### DISCUSSION TITLE
Scala 3 dependency resolution in sbt manager

### DISCUSSION THREAD

**Discussion by ArtyomGabeev** [renovatebot/renovate#29082] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket

### Please tell us more about your question or problem

Starting with scala 3, dependencies are published with major version.
Here is an example:
```
https://mvnrepository.com/artifact/io.circe/circe-core_2.13/0.15.0-M1
https://mvnrepository.com/artifact/io.circe/circe-core_3/0.15.0-M1
```
As you may see, for scala 3 there is only a major version. However if we have scalaVersion defined in build.sbt:
```
ThisBuild / scalaVersion := "3.4.1"
```

renovate bot attempts to resolve dependency with _3.4 prefix, can not find it via sbt manager and fallbacks to maven (which does not work in all cases, e.g. when library pushed without maven support).

Notes:
I can contribute the fix.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Converted to issue: https://github.com/renovatebot/renovate/issues/29087

A PR would be very welcome, thanks.

Usually, if a particular manager or ecosystem changes behavior in a new release then we consider it a _feature request_ to support the new release, and only a _bug fix_ if existing behavior all stopped working. In this case I assume that 2.x -> 2.x upgrades continue working.
